.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,51 @@
+name: Deploy GitHub Pages
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
```

Comment:
Remove `pull_request` from deployment workflow.
Separate workflow to check if the build is successful in the pull request.